File: Chain.Model/Chain.Model/SysRotate.cs
using System;

namespace Chain.Model
{
	[Serializable]
	public class SysRotate
	{
		private int rotateID;

		private string rotateName;

		private DateTime startTime;

		private DateTime endTime;

		private string rotateRemark;

		private int rotateCount;

		private int personTotalCount;

		private int personDayCount;

		private string imageUrl;

		private DateTime createTime;

		private int createUserID;

		private string onePrizeName;

		private int onePrizeCount;

		private int onePrizeWinCount;

		private string twoPrizeName;

		private int twoPrizeCount;

		private int twoPrizeWinCount;

		private string threePrizeName;

		private int threePrizeCount;

		private int threePrizeWinCount;

		private string fourPrizeName;

		private int fourPrizeCount;

		private int fourPrizeWinCount;

		private string fivePrizeName;

		private int fivePrizeCount;

		private int fivePrizeWinCount;

		private string sixPrizeName;

		private int sixPrizeCount;

		private int sixPrizeWinCount;

		private decimal oneRate;

		private decimal twoRate;

		private decimal threeRate;

		private decimal fourRate;

		private decimal fiveRate;

		private decimal sixRate;

		private string rotateRegion;

		private string oneMobile;

		private string twoMobile;

		private string threeMobile;

		private string fourMobile;

		private string fiveMobile;

		private string sixMobile;

		private string oneName;

		private string twoName;

		private string threeName;

		private string fourName;

		private string fiveName;

		private string sixName;

		private int isWinOne;

		public int RotateID
		{
			get
			{
				return this.rotateID;
			}
			set
			{
				this.rotateID = value;
			}
		}

		public string RotateName
		{
			get
			{
				return this.rotateName;
			}
			set
			{
				this.rotateName = value;
			}
		}

		public DateTime StartTime
		{
			get
			{
				return this.startTime;
			}
			set
			{
				this.startTime = value;
			}
		}

		public DateTime EndTime
		{
			get
			{
				return this.endTime;
			}
			set
			{
				this.endTime = value;
			}
		}

		public string RotateRemark
		{
			get
			{
				return this.rotateRemark;
			}
			set
			{
				this.rotateRemark = value;
			}
		}

		public int RotateCount
		{
			get
			{
				return this.rotateCount;
			}
			set
			{
				this.rotateCount = value;
			}
		}

		public int PersonTotalCount
		{
			get
			{
				return this.personTotalCount;
			}
			set
			{
				this.personTotalCount = value;
			}
		}

		public int PersonDayCount
		{
			get
			{
				return this.personDayCount;
			}
			set
			{
				this.personDayCount = value;
			}
		}

		public string ImageUrl
		{
			get
			{
				return this.imageUrl;
			}
			set
			{
				this.imageUrl = value;
			}
		}

		public DateTime CreateTime
		{
			get
			{
				return this.createTime;
			}
			set
			{
				this.createTime = value;
			}
		}

		public int CreateUserID
		{
			get
			{
				return this.createUserID;
			}
			set
			{
				this.createUserID = value;
			}
		}

		public string OnePrizeName
		{
			get
			{
				return this.onePrizeName;
			}
			set
			{
				this.onePrizeName = value;
			}
		}

		public int OnePrizeCount
		{
			get
			{
				return this.onePrizeCount;
			}
			set
			{
				this.onePrizeCount = value;
			}
		}

		public int OnePrizeWinCount
		{
			get
			{
				return this.onePrizeWinCount;
			}
			set
			{
				this.onePrizeWinCount = value;
			}
		}

		public string TwoPrizeName
		{
			get
			{
				return this.twoPrizeName;
			}
			set
			{
				this.twoPrizeName = value;
			}
		}

		public int TwoPrizeCount
		{
			get
			{
				return this.twoPrizeCount;
			}
			set
			{
				this.twoPrizeCount = value;
			}
		}

		public int TwoPrizeWinCount
		{
			get
			{
				return this.twoPrizeWinCount;
			}
			set
			{
				this.twoPrizeWinCount = value;
			}
		}

		public string ThreePrizeName
		{
			get
			{
				return this.threePrizeName;
			}
			set
			{
				this.threePrizeName = value;
			}
		}

		public int ThreePrizeCount
		{
			get
			{
				return this.threePrizeCount;
			}
			set
			{
				this.threePrizeCount = value;
			}
		}

		public int ThreePrizeWinCount
		{
			get
			{
				return this.threePrizeWinCount;
			}
			set
			{
				this.threePrizeWinCount = value;
			}
		}

		public string FourPrizeName
		{
			get
			{
				return this.fourPrizeName;
			}
			set
			{
				this.fourPrizeName = value;
			}
		}

		public int FourPrizeCount
		{
			get
			{
				return this.fourPrizeCount;
			}
			set
			{
				this.fourPrizeCount = value;
			}
		}

		public int FourPrizeWinCount
		{
			get
			{
				return this.fourPrizeWinCount;
			}
			set
			{
				this.fourPrizeWinCount = value;
			}
		}

		public string FivePrizeName
		{
			get
			{
				return this.fivePrizeName;
			}
			set
			{
				this.fivePrizeName = value;
			}
		}

		public int FivePrizeCount
		{
			get
			{
				return this.fivePrizeCount;
			}
			set
			{
				this.fivePrizeCount = value;
			}
		}

		public int FivePrizeWinCount
		{
			get
			{
				return this.fivePrizeWinCount;
			}
			set
			{
				this.fivePrizeWinCount = value;
			}
		}

		public string SixPrizeName
		{
			get
			{
				return this.sixPrizeName;
			}
			set
			{
				this.sixPrizeName = value;
			}
		}

		public int SixPrizeCount
		{
			get
			{
				return this.sixPrizeCount;
			}
			set
			{
				this.sixPrizeCount = value;
			}
		}

		public int SixPrizeWinCount
		{
			get
			{
				return this.sixPrizeWinCount;
			}
			set
			{
				this.sixPrizeWinCount = value;
			}
		}

		public decimal OneRate
		{
			get
			{
				return this.oneRate;
			}
			set
			{
				this.oneRate = value;
			}
		}

		public decimal TwoRate
		{
			get
			{
				return this.twoRate;
			}
			set
			{
				this.twoRate = value;
			}
		}

		public decimal ThreeRate
		{
			get
			{
				return this.threeRate;
			}
			set
			{
				this.threeRate = value;
			}
		}

		public decimal FourRate
		{
			get
			{
				return this.fourRate;
			}
			set
			{
				this.fourRate = value;
			}
		}

		public decimal FiveRate
		{
			get
			{
				return this.fiveRate;
			}
			set
			{
				this.fiveRate = value;
			}
		}

		public decimal SixRate
		{
			get
			{
				return this.sixRate;
			}
			set
			{
				this.sixRate = value;
			}
		}

		public string RotateRegion
		{
			get
			{
				return this.rotateRegion;
			}
			set
			{
				this.rotateRegion = value;
			}
		}

		public string OneMobile
		{
			get
			{
				return this.oneMobile;
			}
			set
			{
				this.oneMobile = value;
			}
		}

		public string TwoMobile
		{
			get
			{
				return this.twoMobile;
			}
			set
			{
				this.twoMobile = value;
			}
		}

		public string ThreeMobile
		{
			get
			{
				return this.threeMobile;
			}
			set
			{
				this.threeMobile = value;
			}
		}

		public string FourMobile
		{
			get
			{
				return this.fourMobile;
			}
			set
			{
				this.fourMobile = value;
			}
		}

		public string FiveMobile
		{
			get
			{
				return this.fiveMobile;
			}
			set
			{
				this.fiveMobile = value;
			}
		}

		public string SixMobile
		{
			get
			{
				return this.sixMobile;
			}
			set
			{
				this.sixMobile = value;
			}
		}

		public string OneName
		{
			get
			{
				return this.oneName;
			}
			set
			{
				this.oneName = value;
			}
		}

		public string TwoName
		{
			get
			{
				return this.twoName;
			}
			set
			{
				this.twoName = value;
			}
		}

		public string ThreeName
		{
			get
			{
				return this.threeName;
			}
			set
			{
				this.threeName = value;
			}
		}

		public string FourName
		{
			get
			{
				return this.fourName;
			}
			set
			{
				this.fourName = value;
			}
		}

		public string FiveName
		{
			get
			{
				return this.fiveName;
			}
			set
			{
				this.fiveName = value;
			}
		}

		public string SixName
		{
			get
			{
				return this.sixName;
			}
			set
			{
				this.sixName = value;
			}
		}

		public int IsWinOne
		{
			get
			{
				return this.isWinOne;
			}
			set
			{
				this.isWinOne = value;
			}
		}
	}
}

File: Backup/ChainStock/Scripts/Module/MicroWebsite/SysRotate.js
$(document).ready(function () {
    $("[class=parent]").click(function () {
        $(this).parent().parent().next().css("display") == "none" ? $(this).parent().parent().next().css("display", "") : $(this).parent().parent().next().css("display", "none");
        $(this).parent().parent().next().css("display") == "none" ? $(this).prev("div").css("background", "url(../Inc/Style/images/plus.gif) no-repeat") : $(this).prev("div").css("background", "url(../Inc/Style/images/minus.gif) no-repeat");
    })
    $("[class=parent]").mouseover(function () {
        $(this).css({ "textDecoration": "underline", "color": "#FF00FE" });
    }).mouseout(function () {
        $(this).css({ "textDecoration": "none", "color": "#AF0081" });
    })

    //文本消息保存按钮点击事件
    $("#btnRotateSave").bind("click", btnRotateSave);
    //文本消息重置按钮点击事件
    $("#btnTextRuleReset").bind("click", btnTextRuleResetClick);

    $("#RotatePhoto_Uploadify").uploadify({
        'uploader': "../images/swf/uploadify.swf",
        'script': "../Service/MicroWebsiteUpload.ashx",
        'cancelImg': "../images/member/cancel.png",
        'folder': "../Upload/MicroWebsite/Rotate",
        'queueID': 'Rotate_fileQueue',
        'buttonImg': "../images/member/selectImg.jpg",
        'height': 25,
        'width': 70,
        'fileExt': "*.jpg;*.jpeg;*.gif;*.png;*.bmp",
        'fileDesc': "请选择格式为GIF、JPG、PNG或BMP的图片",
        'fileDataName': "Rotate",
        'auto': false,
        'multi': false,
        'method': 'get',
        'sizeLimit': 512000,
        'onError': function (event, ID, fileObj, errorObj) {
            if (errorObj.type == "File Size")
                alert("对不起，上传的图片不能超过500K");
            else
                alert(errorObj.type + ' Error: ' + errorObj.info);
        },
        'onComplete': function (event, ID, fileObj, response, data) {
            if (response.length > 1) {
                $("#txtRotatePhoto").val("../Upload/MicroWebsite/Rotate/" + response);            
                $("#imgRotatePhoto").attr("src", "../Upload/MicroWebsite/Rotate/" + response + "?" + GetGuid());
            }
        },
        'onSelect': function (event, queueID, fileObj) {
            $("#RotatePhoto_Uploadify").uploadifySettings('scriptData', { 'name': $("#MerchantPhoto").val() });
        }
    });
})



//文本消息保存按钮点击事件 响应函数
function btnRotateSave() {
    var type = $("#txtRotateID").val() == "" ? "SysRotateAdd" : "SysRotateEdit";
    var txtRotateName = $.trim($("#txtRotateName").val());
    var txtRotateRemark = $.trim($("#txtRotateRemark").val());
    var txtStartTime = $.trim($("#txtStartTime").val());
    var txtEndTime = $.trim($("#txtEndTime").val());
//    var txtRotateCount = $.trim($("#txtRotateCount").val());
//    var txtPersonTotalCount = $.trim($("#txtPersonTotalCount").val());
//    var txtPersonDayCount = $.trim($("#txtPersonDayCount").val());
    var txtOnePrizeName = $.trim($("#txtOnePrizeName").val());
    var txtOnePrizeCount = $.trim($("#txtOnePrizeCount").val());
    var txtTwoPrizeName = $.trim($("#txtTwoPrizeName").val());
    var txtTwoPrizeCount = $.trim($("#txtTwoPrizeCount").val());
    var txtThreePrizeName = $.trim($("#txtThreePrizeName").val());
    var txtThreePrizeCount = $.trim($("#txtThreePrizeCount").val());
    var txtFourPrizeName = $.trim($("#txtFourPrizeName").val());
    var txtFourPrizeCount = $.trim($("#txtFourPrizeCount").val());
    var txtFivePrizeName = $.trim($("#txtFivePrizeName").val());
    var txtFivePrizeCount = $.trim($("#txtFivePrizeCount").val());
    var txtSixPrizeName = $.trim($("#txtSixPrizeName").val());
    var txtSixPrizeCount = $.trim($("#txtSixPrizeCount").val());
    var txtOneRate = $.trim($("#txtOneRate").val());
    var txtTwoRate = $.trim($("#txtTwoRate").val());
    var txtThreeRate = $.trim($("#txtThreeRate").val());
    var txtFourRate = $.trim($("#txtFourRate").val());
    var txtFiveRate = $.trim($("#txtFiveRate").val());
    var txtSixRate = $.trim($("#txtSixRate").val());
    var txtRotateRegion = $.trim($("#txtRotateRegion").val());
    var strErrorMsg = "";

    if (txtRotateName == "") { strErrorMsg += "<li>活动名称不能为空;</li>"; }
    if (txtRotateRemark == "") { strErrorMsg += "<li>活动说明不能为空;</li>"; }
    if (txtStartTime == "") { strErrorMsg += "<li>活动开始时间不能为空;</li>"; }
    if (txtEndTime == "") { strErrorMsg += "<li>活动结束时间不能为空;</li>"; }
    if (Date.parse(txtEndTime) < Date.parse(txtStartTime)) {
        strErrorMsg += "<li>活动结束时间不能小于开始时间;</li>";
    }
    if (txtRotateRegion == "") { strErrorMsg += "<li>活动指令不能为空;</li>"; }
    
//    if (txtRotateCount == "") { strErrorMsg += "<li>活动总人数不能为空;</li>"; }
//    if (txtPersonTotalCount == "") { strErrorMsg += "<li>抽奖总次数不能为空;</li>"; }
//    if (txtPersonDayCount == "") { strErrorMsg += "<li>每天抽奖次数不能为空;</li>"; }
    if (txtOnePrizeName == "") { strErrorMsg += "<li>一等奖奖品名称不能为空;</li>"; }
    if (txtOnePrizeCount == "") { strErrorMsg += "<li>一等奖奖品数量不能为空;</li>"; }
    if (txtTwoPrizeName == "") { strErrorMsg += "<li>二等奖奖品名称不能为空;</li>"; }
    if (txtTwoPrizeCount == "") { strErrorMsg += "<li>二等奖奖品数量不能为空;</li>"; }
    if (txtThreePrizeName == "") { strErrorMsg += "<li>三等奖奖品名称不能为空;</li>"; }
    if (txtThreePrizeCount == "") { strErrorMsg += "<li>三等奖奖品数量不能为空;</li>"; }
    if (txtFourPrizeName == "") { strErrorMsg += "<li>四等奖奖品名称不能为空;</li>"; }
    if (txtFourPrizeCount == "") { strErrorMsg += "<li>四等奖奖品数量不能为空;</li>"; }
    if (txtFivePrizeName == "") { strErrorMsg += "<li>五等奖奖品名称不能为空;</li>"; }
    if (txtFivePrizeCount == "") { strErrorMsg += "<li>五等奖奖品数量不能为空;</li>"; }
    if (txtSixPrizeName == "") { strErrorMsg += "<li>六等奖奖品名称不能为空;</li>"; }
    if (txtSixPrizeCount == "") { strErrorMsg += "<li>六等奖奖品数量不能为空;</li>"; }
    if (txtOneRate == "") { strErrorMsg += "<li>一等中奖几率不能为空;</li>"; }
    if (txtTwoRate == "") { strErrorMsg += "<li>二等奖中奖几率不能为空;</li>"; }
    if (txtThreeRate == "") { strErrorMsg += "<li>三等奖中奖几率不能为空;</li>"; }
    if (txtFourRate == "") { strErrorMsg += "<li>四等奖中奖几率不能为空;</li>"; }
    if (txtFiveRate == "") { strErrorMsg += "<li>五等奖中奖几率不能为空;</li>"; }
    if (txtSixRate == "") { strErrorMsg += "<li>六等奖中奖几率不能为空;</li>"; }
    if (strErrorMsg != "") {
        strErrorMsg = "<div>操作出现以下错误，请核查后重试！</div><ul>" + strErrorMsg + "</ul>";
        art.dialog({
            title: '系统提醒',
            icon: 'error', //图标
            content: strErrorMsg,
            lock: true
        });
        return false;
    }

//    art.dialog({
//        content: "将要" + (type == "SysRotateAdd" ? "添加" : "编辑") + "幸运大转盘活动吗。\n确定操作吗？",
//        lock: true,
//        ok: function () {

//            doAjax("../", type, {
//                "RotateName": txtRotateName,
//                "RotateRemark": txtRotateRemark,
//                "StartTime": txtStartTime,
//                "EndTime": txtEndTime,
//                "RotateCount": txtRotateCount,
//                "PersonTotalCount": txtPersonTotalCount,
//                "PersonDayCount": txtPersonDayCount,
//                "OnePrizeName": txtOnePrizeName,
//                "OnePrizeCount": txtOnePrizeCount,
//                "TwoPrizeName": txtTwoPrizeName,
//                "TwoPrizeCount": txtTwoPrizeCount,
//                "ThreePrizeName": txtThreePrizeName,
//                "ThreePrizeCount": txtThreePrizeCount,
//                "FourPrizeName": txtFourPrizeName,
//                "FourPrizeCount": txtFourPrizeCount,
//                "FivePrizeName": txtFivePrizeName,
//                "FivePrizeCount": txtFivePrizeCount,
//                "SixPrizeName": txtSixPrizeName,
//                "SixPrizeCount": txtSixPrizeCount,
//                "OneRate": txtOneRate,
//                "TwoRate": txtTwoRate,
//                "ThreeRate": txtThreeRate,
//                "FourRate": txtFourRate,
//                "FiveRate": txtFiveRate,
//                "SixRate": txtSixRate,
//                "RotateID": $.trim($("#txtRotateID").val())
//               
//            }, "json", function (json) {
//                alert("");
//                if (json == "0") {
//                    art.dialog({
//                        title: '系统提示',
//                        content: ("系统异常，未保存数据，请再次点击保存！"),
//                        lock: true
//                    })
//                }
//                else if (json == "-2") {
//                    art.dialog({
//                        title: '系统提示',
//                        time: 1.5,
//                        content: ("该时间段有正在进行的大转盘活动，请修改活动时间再保存！"),
//                        lock: true
//                    })
//                }
//                else if (json == "-1") {
//                    art.dialog({
//                        title: '系统提示',
//                        time: 1.5,
//                        content: ("系统错误！"),
//                        lock: true
//                    })
//                }
//                else {
//                    art.dialog({
//                        title: '系统提示',
//                        time: 1.5,
//                        content: '保存成功！',
//                        close: function () { location.href = "SysRotateList.aspx?PID=144"; }
//                    })
//                }
//            })
//        },
//        cancelVal: '取消',
//        cancel: true
//    })
}

//文本消息重置按钮点击事件  响应函数
function btnTextRuleResetClick() {
    var number = parseInt($("#txtNumber").val());
    if (!isNaN(number)) {
        if (number < 0) {
            $("#txtSendContent").val("");
        }
    }
    else {
        $("#txtNumber,#txtDescRule,#txtSendContent").val("");
    }
}


